sn_protocol/src/storage/header.rs

```diff
@@ -123,6 +130,11 @@ mod tests {
 
     #[test]
     fn verify_record_header_encoded_size() -> Result<()> {
+        let record_with_payment = RecordHeader {
+            kind: RecordKind::ChunkWithPayment,
+        }
+        .try_serialize()?;
+        assert_eq!(record_with_payment.len(), RecordHeader::SIZE);
```

Comment:
Can we also add the check for `RegisterWithPayment`. We had some undefined behaviour with a specific variant earlier if I'm not wrong.